Client/Frontend/Settings/SettingsViewController.swift

```diff
@@ -196,6 +197,53 @@ class SettingsViewController: TableViewController {
                     self.navigationController?.pushViewController(clearPrivateData, animated: true)
                 },
                 accessory: .disclosureIndicator
+            ),
+            Row(
+                text: Strings.Block_all_cookies,
+                accessory: .switchToggle(
+                    value: Preferences.Privacy.blockAllCookies.value, { [unowned self] in
+                        func toggleCookieSetting(with status: Bool) {
+                            //Lock/Unlock Cookie Folder
+                            let success = FileManager.default.setFolderAccess([
+                                (.cookie, status),
+                                (.webSiteData, status)
+                                ])
+                            if success {
+                                Preferences.Privacy.blockAllCookies.value = status
+                            } else {
+                                //Revert the changes. Not handling success here to avoid a loop.
+                                _ = FileManager.default.setFolderAccess([
+                                    (.cookie, false),
+                                    (.webSiteData, false)
+                                    ])
+                                self.toggleSwitch(on: false, section: self.privacySection, rowUUID: Preferences.Privacy.blockAllCookies.key)
+                                
+                                // TODO: Throw Alert to user to try again?
+                                let alert = UIAlertController(title: nil, message: Strings.Block_all_cookies_failed_alert_msg, preferredStyle: .alert)
+                                alert.addAction(UIAlertAction(title: Strings.OKString, style: .default))
+                                self.present(alert, animated: true)
+                            }
+                        }
+                        if $0 {
+                            let status = $0
+                            // THROW ALERT to inform user of the setting
+                            let alert = UIAlertController(title: Strings.Block_all_cookies_alert_title, message: Strings.Block_all_cookies_alert_info, preferredStyle: .alert)
+                            let okAction = UIAlertAction(title: Strings.Block_all_cookies_action, style: .destructive, handler: { (action) in
+                                toggleCookieSetting(with: status)
+                            })
+                            alert.addAction(okAction)
+                            
+                            let cancelAction = UIAlertAction(title: Strings.CancelButtonTitle, style: .cancel, handler: { (action) in
+                                self.toggleSwitch(on: false, section: self.privacySection, rowUUID: Preferences.Privacy.blockAllCookies.key)
```

Comment:
You should be able to use `row.view` since I changed the implementation of `BoolRow`'s in https://github.com/brave/brave-ios/pull/743/files and it will return the actual `UISwitch` instead of creating a new one each time.

Reply:
Great, will update

Reply:
Updated the BoolRow class to have take a custom `on Value Change` closure as param.